packages/SwingSet/src/kernel/vatManager.js

```diff
@@ -282,19 +282,27 @@ export default function makeVatManager(
 
   // dispatch handlers: these are used by the kernel core
 
-  function doProcess(d, errmsg) {
-    const dispatchOp = d[0];
-    const dispatchArgs = d.slice(1);
-    transcriptStartDispatch(d);
-    return process(
-      () => dispatch[dispatchOp](...dispatchArgs),
-      () => transcriptFinishDispatch(),
-      err => {
-        if (errmsg !== null) {
-          console.log(`doProcess: ${errmsg}:`, err);
-        }
-      },
-    );
+  function runAndWait(f, errmsg) {
+    Promise.resolve()
+      .then(f)
+      .then(undefined, err => console.log(`doProcess: ${errmsg}:`, err));
```

Comment:
let's add a note here to remind us to not make the mistake I made yesterday.. something like:

Note that we explicitly drop the Promise returned by f(), after attaching an error-logging function. The kernel waits for the vat to become quiescent (nothing on the promise queue). The vat is not given access to the timer or IO queues (no setImmediate or setInterval), therefore once the promise queue is empty, the vat has lost "agency" (the ability to initiate further execution). The kernel does *not* wait for the dispatch handler's return Promise to resolve: the kernel must be defensive against a malicious vat, and it might never resolve that Promise.

Reply:
Actually, it's a good opportunity to document what `runAndWait` actually does overall.  Done.